README.md

```diff
@@ -4,12 +4,44 @@ A Node JS implementation of the CSS Object Model [CSSStyleDeclaration interface]
 
 [![NpmVersion](https://img.shields.io/npm/v/cssstyle.svg)](https://www.npmjs.com/package/cssstyle) [![Build Status](https://travis-ci.org/jsdom/cssstyle.svg?branch=master)](https://travis-ci.org/jsdom/cssstyle) [![codecov](https://codecov.io/gh/jsdom/cssstyle/branch/master/graph/badge.svg)](https://codecov.io/gh/jsdom/cssstyle)
 
----
+## Background
 
-#### Background
+This package is an extension of the CSSStyleDeclaration class in Nikita Vasilyev's [CSSOM](https://github.com/NV/CSSOM) with added support for CSS 2 & 3 properties. The primary use case is for testing browser code in a Node environment.
 
-This package is an extension of the CSSStyleDeclaration class in Nikita Vasilyev's [CSSOM](https://github.com/NV/CSSOM) with added support for CSS 2 & 3 properties. The primary use case is for testing browser code in a Node environment. 
-
-It was originally created by Chad Walker, it is now maintaind by Jon Sakas and other open source contributors.
+It was originally created by Chad Walker, it is now maintained by Jon Sakas and other open source contributors.
 
 Bug reports and pull requests are welcome.
+
+## APIs
+
+This package exposes two flavors of the `CSSStyleDeclaration` interface depending on the imported module.
+
+### `cssstyle` module
+
+This module default-exports the `CSSStyleDeclaration` interface constructor, with the change that it can be constructed and takes an `onChangeCallback` optional parameter:
+
+```ts
+export = class CSSStyleDeclaration {
+  /**
+   * @param onChangeCallback The callback that is invoked whenever a property changes.
+   */
+  constructor(onChangeCallback?: ((cssText: string) => void) | null);
+}
+```
```

Comment:
Can we stick to JavaScript? It's strange to have TypeScript in the README of a JavaScript project, and also I think more words to explain what this is is better than fewer.

```suggestion
This module default-exports the `CSSStyleDeclaration` interface constructor, with the change that it can be constructed with an optional `onChangeCallback` parameter. Whenever any CSS property is modified through an instance of this class, the callback (if provided) will be called with a string that represents all CSS properties of this element, serialized. This allows the embedding environment to properly reflect the style changes to an element's `style` attribute.

Here is a crude example of using the `onChangeCallback` to implement the `style` property of `HTMLElement`:
```js
const CSSStyleDeclaration = require('cssstyle');

class HTMLElement extends Element {
  constructor() {
    this._style = new CSSStyleDeclaration(newCSSText => {
      this.setAttributeNS(null, "style", newCSSText);
    });
  }
  get style() {
    return this._style;
  }
  set style(text) {
    this._style.cssText = text;
  }
}
<!-- the suggestion block doesn't let me type this: ``` -->
```

Reply:
For future reference, you can do that by using more than 3 ````​```​```` characters:

`````markdown
````suggestion
This module default-exports the `CSSStyleDeclaration` interface constructor, with the change that it can be constructed with an optional `onChangeCallback` parameter. Whenever any CSS property is modified through an instance of this class, the callback (if provided) will be called with a string that represents all CSS properties of this element, serialized. This allows the embedding environment to properly reflect the style changes to an element's `style` attribute.

Here is a crude example of using the `onChangeCallback` to implement the `style` property of `HTMLElement`:
```js
const CSSStyleDeclaration = require('cssstyle');

class HTMLElement extends Element {
  constructor() {
    this._style = new CSSStyleDeclaration(newCSSText => {
      this.setAttributeNS(null, "style", newCSSText);
    });
  }
  get style() {
    return this._style;
  }
  set style(text) {
    this._style.cssText = text;
  }
}
```
````
`````